Multiple TextField support

The current `textModel` is only working when there is one TextField displayed.

Reproduce the bug by using the [simpleDemo](https://github.com/Drakirus/go-flutter-desktop-embedder/tree/master/example/simpleDemo) example. The embedder crash when words are typed in a TextField when there is another TextField displayed on the screen that has been filled.  

```flutter: Observatory listening on http://127.0.0.1:37183/
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/Drakirus/go-flutter-desktop-embedder.(*textModel).addChar(0x634080, 0xc0000a9c74, 0x1, 0x1)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/textModel.go:32 +0x15e
github.com/Drakirus/go-flutter-desktop-embedder.glfwCharCallback(0xc0000b6000, 0x66)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/gutter.go:198 +0x76
github.com/go-gl/glfw/v3.2/glfw.goCharCB(0xef9990, 0x66)
        /home/drakirus/lab/go/src/github.com/go-gl/glfw/v3.2/glfw/input.go:294 +0x51
github.com/go-gl/glfw/v3.2/glfw._cgoexpwrap_f7a05a868eb7_goCharCB(0xef9990, 0x66)
        _cgo_gotypes.go:2147 +0x33
github.com/go-gl/glfw/v3.2/glfw._Cfunc_glfwPollEvents()
        _cgo_gotypes.go:1196 +0x41
github.com/go-gl/glfw/v3.2/glfw.PollEvents()
        /home/drakirus/lab/go/src/github.com/go-gl/glfw/v3.2/glfw/window.go:777 +0x20
github.com/Drakirus/go-flutter-desktop-embedder.Run(0xc0000a9f70, 0x3, 0x3, 0x0, 0x0)
        /home/drakirus/lab/go/src/github.com/Drakirus/go-flutter-desktop-embedder/gutter.go:80 +0x21d
main.main()
        /home/drakirus/lab/go/src/go-flutter-desktop-embedder/example/demo/main.go:25 +0xc6
```

--- a/flutter/message.go
+++ b/flutter/message.go
@@ -15,9 +15,9 @@ const (
 	TextUpdateStateMethod = "TextInputClient.updateEditingState"
 
 	// text
-	SetClientMethod       = "TextInput.setClient"
-	ClearClientMethod     = "TextInput.clearClient"
-	SetEditingStateMethod = "TextInput.setEditingState"
+	TextInputClientSet    = "TextInput.setClient"
+	TextInputClientClear  = "TextInput.clearClient"
+	TextInputSetEditState = "TextInput.setEditingState"
 )
 
 // Message is the json content of a PlatformMessage

--- a/gutter.go
+++ b/gutter.go
@@ -271,26 +271,24 @@ func onPlatformMessage(platMessage flutter.PlatformMessage, window unsafe.Pointe
 	}
 
 	if platMessage.Channel == flutter.TextInputChannel {
-
-		if message.Method == flutter.ClearClientMethod {
+		switch message.Method {
+		case flutter.TextInputClientClear:
 			state.clientID = 0
-		}
-
-		if message.Method == flutter.SetClientMethod {
+		case flutter.TextInputClientSet:
 			var body []interface{}
 			json.Unmarshal(message.Args, &body)
 			state.clientID = body[0].(float64)
-		}
-
-		if message.Method == flutter.SetEditingStateMethod {
+		case flutter.TextInputSetEditState:
 			if state.clientID != 0 {
 				editingState := flutter.ArgsEditingState{}
 				json.Unmarshal(message.Args, &editingState)
 				state.word = editingState.Text
+				state.selectionBase = editingState.SelectionBase
+				state.selectionExtent = editingState.SelectionExtent
 			}
-
+		default:
+			// log.Printf("unhandled text input method: %#v\n", platMessage.Message)
 		}
-
 	}
 
 	return true

